Mathlib/Data/Nat/Choose/Lucas.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2023 Gareth Ma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gareth Ma
+-/
+import Mathlib.Data.ZMod.Basic
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+# Lucas's theorem
+
+This file contains a proof of [Lucas's theorem](https://en.wikipedia.org/wiki/Lucas's_theorem) about
+binomial coefficients, which says that for primes `p`, `n` choose `k` is congruent to product of
+`n_i` choose `k_i` modulo `p`, where `n_i` and `k_i` are the base-`p` digits of `n` and `k`,
+respectively.
+
+## Main statements
+
+* `lucas_theorem`: the binomial coefficient `n choose k` is congruent to the product of `n_i choose
+k_i` modulo `p`, where `n_i` and `k_i` are the base-`p` digits of `n` and `k`, respectively.
+-/
+
+open Finset hiding choose
+
+open Nat BigOperators Polynomial
+
+namespace Choose
+
+variable {n k p : ℕ} [Fact p.Prime]
+
+/-- For primes `p`, `choose n k` is congruent to `choose (n % p) (k % p) * choose (n / p) (k / p)`
+modulo `p`. -/
+theorem choose_modEq_choose_mod_mul_choose_div :
+    choose n k ≡ choose (n % p) (k % p) * choose (n / p) (k / p) [ZMOD p] := by
```

Comment:
Both sides of this are nats, right? So maybe it'd be nice to have versions using MOD p as well?

Reply:
I agree with Bhavik here we should have the Nat version.

@grhkm21, would you mind either adding this, or simply saying "sorry, can someone do it in a subsequent PR", and then you can merge?

Reply:
How should I name the theorems? I will add some `_nat` and `_int`, but alternatively I can call it `zmod` vs `mod`, but that seems unpopular in Mathlib.

Reply:
`modEq` and `intModEq`, maybe?